Sources/PackageLoading/ToolsVersionLoader.swift

```diff
@@ -88,30 +88,201 @@ extension Manifest {
+        public let leadingWhitespace: Substring
+        /// The Swift tools version specification represented in its constituent parts.
+        public let toolsVersionSpecificationComponents: ToolsVersionSpecificationComponents
+        /// The remaining contents of the manifest that follows right after the tools version specification line.
+        public let contentsAfterToolsVersionSpecification: Substring
+        /// A Boolean value indicating whether the manifest represented in its constituent parts is backward-compatible with Swift ≤ 5.3.
+        public var isCompatibleWithPreSwift5_3_1: Bool {
+            leadingWhitespace.allSatisfy { $0 == "\n" } && toolsVersionSpecificationComponents.isCompatibleWithPreSwift5_3_1
+        }
+    }
+    
+    /// A representation of a Swift tools version specification in its constituent parts.
+    ///
+    /// A Swift tools version specification consists of the following parts:
+    ///
+    ///     //  swift-tools-version:  5.3.1
+    ///     ⌃~⌃~⌃~~~~~~~~~~~~~~~~~~~⌃~⌃~~~~
+    ///     │ │ └ label             │ └ version specifier
+    ///     │ └ spacing             └ spacing
+    ///     └ comment marker
+    ///
+    public struct ToolsVersionSpecificationComponents {
+        /// The comment marker.
+        ///
+        /// In a well-formed Swift tools version specification, the comment marker is `"//"`.
+        public let commentMarker: Substring
+        
+        /// The spacing after the comment marker
+        ///
+        /// In a well-formed Swift tools version specification, the spacing after the comment marker is a continuous sequence of horizontal whitespace characters.
+        ///
+        /// For Swift ≤ 5.3, the spacing after the comment marker must be a single `U+0020`.
+        public let spacingAfterCommentMarker: Substring
+        
+        /// The label part of the Swift tools version specification.
+        ///
+        /// In a well-formed Swift tools version specification, the label is `"swift-tools-version:"`
+        public let label: Substring
+        
+        /// The spacing between the label part of the Swift tools version specification and the version specifier.
+        ///
+        /// In a well-formed Swift tools version specification, the spacing after the label is a continuous sequence of horizontal whitespace characters.
+        ///
+        /// For Swift ≤ 5.3, no spacing is allowed after the label.
+        public let spacingAfterLabel: Substring
+        
+        /// The version specifier.
+        public let versionSpecifier: Substring
+        
+        /// A Boolean value indicating whether everything up to the version specifier in the Swift tools version specification represented in its constituent parts is well-formed.
+        public var everythingUpToVersionSpecifierIsWellFormed: Bool {
+            // FIXME: Make `label` case sensitive?
```

Comment:
It probably should be case sensitive.  Is there a strong reason to allow case-insensitivity?

Reply:
It's case-insensitive because the `regex` on the main branch is [case insensitive](https://github.com/WowbaggersLiquidLunch/swift-package-manager/blob/49cfc46bc5defd3ce8e0c0261e3e2cb475bcdb91/Sources/PackageLoading/ToolsVersionLoader.swift#L183). There is also a [test](https://github.com/WowbaggersLiquidLunch/swift-package-manager/blob/49cfc46bc5defd3ce8e0c0261e3e2cb475bcdb91/Tests/PackageLoadingTests/ToolsVersionLoaderTests.swift#L45) for the case insensitivity.

I can make it case sensitive, and have SwiftPM throw an error if the user used a wrong casing. It might be source-breaking for some existing manifests, though.

Reply:
Ah, got it.  No, I was only reacting because I thought you were proposing a change.  In general, I think that the fewer changes to the semantics, the better.  Some my previous comments have perhaps been made under wrong understanding about what was currently allowed.

Reply:
In 70ce4c36a9a0a3cbf4a9e8b83c592d067f630faf, I replaced the `FIXME` callout with an explanation of why it is case-insensitive. The explanation also serves to notify future readers of the case-insensitivity, because it might be overlooked.

Reply:
Thanks!